src/ModelContextProtocol/Server/McpServer.cs

```
@@ -140,6 +147,18 @@ options.GetCompletionHandler is { } handler ?
                 (request, ct) => Task.FromResult(new CompleteResult() { Completion = new() { Values = [], Total = 0, HasMore = false } }));
     }
 
+    private void AddLoggingLevelNotificationHandler(McpServerOptions options)
```

Sorry for the brief review comment earlier. I was in transit, and thought I had enough time to elaborate and I didn't.

This should be set based on "notifications/message". It should be set based in the "logging/setLevel" request handler.

https://spec.modelcontextprotocol.io/specification/2025-03-26/server/utilities/logging/

The notification handler only makes sense client side.

I wonder if we should actually throw an exception if registering a notification handler on the wrong side of Client/Server, in the cases where the spec clearly defines where to handle it?
@stephentoub